textacy/spacy_utils.py

```diff
@@ -90,28 +90,15 @@ def normalized_str(token):
         raise TypeError(msg)
 
 
-def merge_spans(spans, doc):
+def merge_spans(spans):
     """
-    Merge spans so that each only takes up a single token. Since each Span is
-    a view into the doc, merging one invalidates the others. (This will get fixed!)
-    The temporary solution is to gather the information first, before merging.
-    NB: This modifies `doc` in-place!
+    Merge spans *in-place* within parent doc so that each takes up a single token.
```

Comment:
No. A span is a consecutive sequence of tokens, and this function allows you to merge such sequences into a single token. So, Span("the quick brown fox") = [Token("the"), Token("quick"), Token("brown"), Token("fox")] becomes Token("the quick brown fox").